osu.Game/Overlays/UserProfileOverlay.cs

```diff
@@ -19,65 +20,73 @@ namespace osu.Game.Overlays
 {
     public class UserProfileOverlay : FullscreenOverlay
     {
-        private ProfileSection lastSection;
-        private ProfileSection[] sections;
         private GetUserRequest userReq;
         protected ProfileHeader Header;
         private ProfileSectionsContainer sectionsContainer;
         private ProfileTabControl tabs;
 
+        private CancellationTokenSource cancellationSource;
+
         public const float CONTENT_X_MARGIN = 70;
 
+        public UserProfileOverlay()
+        {
+            Add(new Box
+            {
+                RelativeSizeAxes = Axes.Both,
+                Colour = OsuColour.Gray(0.1f)
+            });
+        }
+
         public void ShowUser(long userId) => ShowUser(new User { Id = userId });
 
-        public void ShowUser(User user, bool fetchOnline = true)
+        public void ShowUser(User user)
         {
+            cancellationSource?.Cancel();
+            cancellationSource = null;
+
+            userReq?.Cancel();
+            userReq = null;
+
             if (user == User.SYSTEM_USER) return;
 
             Show();
 
+            // we are already showing the correct user.
             if (user.Id == Header?.User.Value?.Id)
                 return;
 
-            userReq?.Cancel();
-            Clear();
-            lastSection = null;
-
-            sections = new ProfileSection[]
-            {
-                //new AboutSection(),
-                new RecentSection(),
-                new RanksSection(),
-                //new MedalsSection(),
-                new HistoricalSection(),
-                new BeatmapsSection(),
-                new KudosuSection()
-            };
+            // temporary until this class supports proper bindable flow (like BeatmapSetOverlay).
+            sectionsContainer?.FadeOut(200);
+            sectionsContainer?.Expire();
 
-            tabs = new ProfileTabControl
-            {
-                RelativeSizeAxes = Axes.X,
-                Anchor = Anchor.TopCentre,
-                Origin = Anchor.TopCentre,
-                Height = 30
-            };
-
-            Add(new Box
-            {
-                RelativeSizeAxes = Axes.Both,
-                Colour = OsuColour.Gray(0.1f)
-            });
+            userReq = new GetUserRequest(user.Id);
+            userReq.Success += userLoadComplete;
```

Comment:
Not sure if the request fail should not be handled. This is only called once per user, which would result in an incomplete view for the user.

Reply:
it would result in an empty view